### ISSUE TITLE
snes9x v1.53 Super punch out and Hook emulation issues

### ISSUE THREAD

**Issue by joepogo** [snes9xgit/snes9x#122]
On snes9x v1.53 hook and super punch out have some serious issues going on that were not there before. 

With Super punch out the game will crash when fighting mad clown, after he does the brain squash the second time the game will hang and become unplayable. 

Evidenced here:

http://www.snes9x.com/phpbb3/viewtopic.php?f=6&t=5009

With hook, everything is OK except for cutscenes between levels, there are numbers and letters floating through the screen where the text should be. Tested multiple known good dumps all with the same issue. Using an older version of snes9x did the trick though so something got messed up in the last update. 

Please consider fixing these 2 games, snes9x is a wonderful emulator!

**Comment by bearoso** [comment]
Not a fan of punch out, and I'd suck terribly at it so as to be unable to get to the appropriate part to fix it. A save state is no help because it saves too much and wouldn't work for debugging. 

I can look at the hook issue if it's easy to get to.

**Comment by bearoso** [comment]
Hook cutscenes look fine in 1.54 on Gtk+ port. See attached. May be a compiler bug in the Windows port?
![hook usa 000](https://cloud.githubusercontent.com/assets/16159164/19167211/43603880-8bd0-11e6-9f57-7b486aee453f.png)

Will try to game-genie to mad clown, but I'm totally lost with Punch-Out!

**Comment by bearoso** [comment]
Was able to reproduce the punch-out issue with a SRAM found on the internet. Takes a random number of brain squashes to crash, though. Might be the copy protection. Might be an old DMA problem.

**Comment by OV2** [comment]
The Hook issue only occurs if you allow invalid vram access (in the windows config file it's the BlockInvalidVRAMAccess setting). If you set it to true (the default) the issue disappears.

**Comment by bearoso** [comment]
OK, so then it's a non issue.

**Comment by joepogo** [comment]
Thanks guys for the responses. Hmm, so perhaps in the windows code i need to check before compiling i assume. Perhaps i did that by accident. 

I assume the super punch out issue is going to be a little harder to fix eh? :) I appreciate your efforts OV and Bearoso. Super punch out had copy protection?

**Comment by bearoso** [comment]
It's in your config file, not the source. You only really need to set BlockInvalidVRAMAccess to false for certain hacks, like the Bahamut Lagoon translation.

I think the super punch out issue popped up when zones overhauled DMA. As of now, DMA is still pretty archaic, and I'd rather it be overhauled instead of adding a hack, but not, at the least, until after the new sound core gets released.

**Comment by joepogo** [comment]
Thanks bearoso for the response. One question, what if you are using a console port like snes9x on wii or xbox, and there is no config file to edit and this problem is popping up with this issue? I just checked on the consoles and there is no config file to edit for those...

One other thing, is the new sound core more or less resource intensive? I dont have a rig that can run byus stuff, id rather stick to snes9x. :)

**Comment by bearoso** [comment]
As OV2 says, that setting should be true by default, as the only thing that requires it is a certain bahamut lagoon patch, and it breaks many more things. Tell the port author to enable it.

The sound core is slightly more intensive, but it's adapted to use less CPU power. The Wii version is already too slow for 1.53.

**Comment by joepogo** [comment]
Weird, i checked and im pretty sure it is true by default. Still showing green Q's flying across the screen in cutscenes.

**Comment by OV2** [comment]
Are you sure you are checking the correct config file? Also, for other platforms than windows the setting is called "AllowInvalidVRAMAccess" in the "Hack" section. And make sure you are not running snes9x with the -invalidvramaccess parameter.
I just verified it again, and with invalid vram access blocked all five variants of the ROM I have played the first cutscene fine, when they are not blocked you get the flying letters. The japanese version has the worst corruption.

**Comment by joepogo** [comment]
Hey OV and bearoso, i appreciate the input. I was able to fix the hook issue, it was an issue on my end. Im pretty sure the punch out issue however is out of my hands and scope of knowledge. :)

One quick question, is the problem with punch out among the same issue as the one that plagues speedy gonzalez freezing when you hit the button on the floor? Or is that some sort of unemulated issue? Sorry, im just a noob. :P

**Comment by jcdenton2k** [comment]
SNES9X thankfully doesn't force accuracy over performance.  If anything, options like PCSX2 that let the end-user choose between accuracy and performance are ideal IMHO.

Regarding Speedy Gonzalez.  That is a rare edge-case hardware issue.  
Source @ byuu : https://archive.is/rrZLY

> Let's take the case of Speedy Gonzales. This is an SNES platformer with no save functionality, and it's roughly 2-3 hours long. At first glance, it appears to run fine in any emulator. Yet once you reach stage 6-1, you can quickly spot the difference between an accurate emulator and a fast one: there is a switch, required to complete the level, where the game will deadlock if a rare hardware edge case is not emulated. One can imagine the frustration of instantly losing three hours of progress and being met with an unbeatable game. Unless the software does everything in the exact same way the hardware used to, the game remains broken.

> Or consider Air Strike Patrol, where a shadow is drawn under your aircraft. This is done using mid-scanline raster effects, which are extraordinarily resource intensive to emulate. But without the raster effects, your aircraft's shadow will not show up, as you see in the screenshot below. It's easy to overlook, especially if you do not know that it is supposed to be there. But once you actually see it, you realize that it's quite helpful. Your aircraft has the ability to drop bombs, and this shadow acts as a sort of targeting system to determine where they will land.—something that's slightly more difficult without this seemingly minor effect.

**Comment by joepogo** [comment]
Thanks for posting that. Hopefully one day these will get added to snes9x one day. :)

Also, one other question, if you look at memmap.cpp you will see code for specific hacks that might benefit console or pc users. 

`/* DON'T RENDER SUBSCREEN */

		/* Don't render subscreen for games that don't switch to high-res mode 
		   in-game or don't use hi-res at all - and yet still need to do stuff 
		   with the subscreen FPS improvement ranges from 7/10fps to 20/25fps 
		   depending on the game ( +25/30fps in some occassions - benchmarked 
		   on PC).

			TODO: This glitches with rewind enabled (RetroArch), force-disable it
			when rewind is active */
		if(
				MATCH_NA("Super Metroid") 	/* Super Metroid*/
				|| MATCH_NA("DONKEY KONG COUNTRY") /* Donkey Kong Country 1 */
				|| MATCH_NA("AREA88")		/* Area 88 */
				|| MATCH_NA("U.N.Squadron")	/* UN Squadron */
				|| MATCH_NA("Perfect Eleven")	/* Perfect Eleven - ISS1 */
				|| MATCH_ID("ADNE")			/* Donkey Kong Country 2 (US) */
				|| MATCH_ID("AWJ")		/* ISS Deluxe / World Soccer 2 Fighting Eleven */
				|| MATCH_NA("JIM POWER THE LOST DIO") /* Jim Power The Lost Dimension */
				|| MATCH_ID("A3C")		/* Donkey Kong Country 3*/
				|| MATCH_NA("Secret of MANA") /* Secret of Mana */
				|| MATCH_NA("Street Fighter2 Turbo1") /* Street Fighter II Turbo */
				|| MATCH_NA("Super Street Fighter21") /* Super Street Fighter II (US) */
				|| MATCH_ID("A2J")		/* Jurassic Park 2 */
				|| MATCH_ID("AJL")		/* Justice League */
				|| MATCH_NA("JOE AND MAC")	/* Joe and Mac */
				|| MATCH_NA("JOE & MAC 2")	/* Joe & Mac 2 */
				|| MATCH_ID("J3")		/* Joe & Mac 3 EU) */
			|| MATCH_NA("JYUTEI SENKI")	/* Jyutei Senki */
			|| MATCH_ID("CQ  ")		/* Stunt Race FX */
#if defined(_XBOX1) || defined(GEKKO)
			|| MATCH_ID("AD8")		/* Doom */
			|| MATCH_ID("A3D")			/* Seiken Densetsu 3 */
			|| MATCH_NA("SeikenDensetsu3Sample1")	/* Seiken Densetsu 3 */
			|| MATCH_NA("ROMANCING SAGA3")		/* Romancing Saga 3 */
#endif
			|| MATCH_NA("T.M.H.T.4")	/* Teenage Mutant Hero Turtles 4 - Turtles In Time (E) */
			|| MATCH_NA("T.M.N.T.")		/* Teenage Mutant Ninja Turtles 4 - Turtles In Time (J) */
			|| MATCH_NA("T.M.N.T.4")	/* Teenage Mutant Ninja Turtles 4 - Turtles In Time (AU) */
			|| MATCH_NA("T.M.N.T. 4")	/* Teenage Mutant Ninja Turtles 4 - Turtles In Time (US) */
			|| MATCH_NA("T.M.N.T. MW")	/* Teenage Mutant Ninja Turtles - Mutant Warriors (J) */
			|| MATCH_NA("T.M.H.T TF")	/* Teenage Mutant Hero Turtles - Tour Fighters (EU) */
			|| MATCH_NA("T.M.N.T. TF")	/* Teenage Mutant Ninja Turtles - Tour Fighters (AU) */
			|| MATCH_NA("T.M.N.T.5")	/* Teenage Mutant Ninja Turtles - Tour Fighters (U) */
			|| MATCH_NA("SUPER CASTLEVANIA 4") /* Super Castlevania IV */
			|| MATCH_NA("AKUMAJO DRACULA") /* Super Castlevania IV */
			|| MATCH_ID("ATVE")		/* Tales of Phantasia (EN) (DeJap)*/
			|| MATCH_NA("MEGAMAN X")		/* Mega Man X (US) */
			|| MATCH_ID("A7R")		/* Rockman / Mega Man 7 */
			|| MATCH_NA("F-ZERO")		/* Rockman / Mega Man 7 */
			|| MATCH_NA("EQUINOX")		/* Equinox */
			|| MATCH_NA("SUPER MARIO KART") /* Super Mario Kart */
			|| MATCH_ID("AR3")		/* Mega Man X3 (JP/US) */
			|| MATCH_NA("SECRET OF EVERMORE") 	/* Secret of Evermore*/
			|| MATCH_ID("AKL")		/* Killer Instinct*/
			|| MATCH_NA("NBA JAM")		/* NBA Jam */
			|| MATCH_NA("PILOTWINGS")	/* Pilotwings */
			|| MATCH_NA("FINAL FIGHT")	/* Final Fight */
			|| MATCH_NA("FINAL FANTASY 6")	/* Final Fantasy VI (JP)*/
			|| MATCH_NA("FINAL FANTASY 3")	/* Final Fantasy III (US)*/
			|| MATCH_NA("ILLUSION OF GAIA USA") /* Illusion of Gaia (US)*/
			|| MATCH_NA("GAIA GENSOUKI 1 JPN")	/* Gaia Gensouki (JPN)*/
			|| MATCH_ID("AEJ") 		/* Earthworm Jim*/
			|| MATCH_ID("YI")		/* Yoshi's Island*/
			|| MATCH_ID("AFJ")		/* Kirby's Dreamland 3 */
			|| MATCH_NA("SUPER MARIOWORLD")	/* Super Mario World*/
			|| MATCH_NA("THE LEGEND OF ZELDA")	/* Zelda 3*/
			|| MATCH_NA("LA LEGENDE DE ZELDA")	/* Zelda 3 (FR)*/
			|| MATCH_NA("ZELDANODENSETSU")		/* Zelda 3 (JPN)*/
			|| MATCH_NA("SPARKSTER")		/* Sparkster*/
			|| MATCH_NA("SIMCITY")		/* Sim City */
			|| MATCH_NA("T2 ARCADE")	/* T2 Arcade */
			|| MATCH_NA("PLOK")		/* Plok!*/
			|| MATCH_NA("ALADDIN")		/* Aladdin */
			|| MATCH_NA("ActRaiser-2 JPN")	/* ActRaiser 2 (JPN) */
			|| MATCH_NA("ActRaiser-2 USA")	/* ActRaiser 2 (US) */
			|| MATCH_ID("A8")		/* ActRaiser 2 (EU) */
			|| MATCH_ID("APT")		/* Pac In Time */
			|| MATCH_ID("AXG")		/*NBA Hangtime */
			|| MATCH_NA("THE CHAOS ENGINE")	/* Chaos Engine */
			|| MATCH_ID("AWD")		/* Weaponlord */
			|| MATCH_NA("TALES OF PHANTASIA")       /* Tales of Phantasia */
			|| MATCH_NA("GS MIKAMI")		/* GS Mikami - Joreishi wa Nice Body*/
			|| MATCH_ID("ATQ")		/* Theme Park */
			|| MATCH_NA("TOTAL CARNAGE")	/* Total Carnage */
			|| MATCH_ID("AJOJ")		/* Jikkyou Oshaberi Parodius*/
			|| MATCH_NA("ROCK N' ROLL RACING") /* Rock 'n Roll Racing */
			|| MATCH_NA("RISE OF THE ROBOTS")	/* Rise Of The Robots*/
			|| MATCH_ID("A3T")		/* Top Gear 3000 */
			|| MATCH_NA("MORTAL KOMBAT")	/* Mortal Kombat I*/
			|| MATCH_NA("MORTAL KOMBAT II")	/* Mortal Kombat II*/
			|| MATCH_NA("STAR FOX")		/* Star Fox (US/JP)*/
			|| MATCH_ID("ST2")		/* Star Fox 2 */
			|| MATCH_NA("STAR WING")		/* Star Wing (EU)*/
			|| MATCH_NA("TINYTOON adventures") /* Buster Busts Loose */
			|| MATCH_ID("3Z")		/* Demon's Crest*/
			|| MATCH_NA("AXELAY")		/* Axelay*/
			|| MATCH_NA("JAKI CRUSH")	/* Jaki Crush */
			|| MATCH_NA("ZOMBIES")		/* Zombies (EU)*/
			|| MATCH_NA("THE JUNGLE BOOK")	/* Jungle Book */
			|| MATCH_NA("ZOMBIES ATE MY NEIGHB")	/* Zombies Ate My Neighbors*/
			|| MATCH_NA("UNIRACERS")		/* Uniracers*/
			|| MATCH_NA("UNIRALLY")		/* Unirally*/
			|| MATCH_NA("CHRONO TRIGGER")	/* Chrono Trigger*/
			|| MATCH_NA("JURASSIC PARK")	/* Jurassic Park*/
			|| MATCH_NA("THE MAGICAL QUEST")	/* The Magical Quest*/
			|| MATCH_NA("SOULBLAZER - 1 USA")	/* Soul Blazer (US)*/
			|| MATCH_NA("SOULBLAZER - 1 ENG")	/* Soul Blazer (PAL)*/
			|| MATCH_NA("SOULBLADER - 1")		/* Soul Blader*/
			|| MATCH_NA("GOKUJYOU PARODIUS")		/* Gokujou Parodius*/
			|| MATCH_ID("ABT")		/* Adventures of Batman and Robin*/
			|| MATCH_ID("5M")		/* Mario All-Stars + World*/
			|| MATCH_NA("SUPER MARIO ALL_STARS") /* Super Mario All-Stars (EU/US)*/
			|| MATCH_NA("SUPERMARIO COLLECTION") /* Super Mario Collection (JP)*/
			|| MATCH_ID("4Q")		/* Super Punch-Out*/
			|| MATCH_NA("HARVEST MOON")	/* Harvest Moon*/
			|| MATCH_ID("ADZ")		/* Dracula X*/
			|| MATCH_ID("A3M")		/* Mortal Kombat 3*/
			|| MATCH_ID("AM4J")		/* Do-Re-Mi Fantasy - Milon no Dokidoki Daibouken*/
			|| MATCH_NA("BT IN BATTLEMANIACS")	/* Battletoads in Battlemaniacs*/
			|| MATCH_NA("SPACE MEGAFORCE")	/* Space Megaforce (US)*/
			|| MATCH_ID("ARX")		/* Mega Man X2 */
			|| MATCH_NA("SUPER ALESTE")	/* Super Aleste (EU/JP)*/
			|| MATCH_NA("VALKEN")		/* Assault Suits Valken (JP)*/
			|| MATCH_NA("CYBERNATOR")	/* Cybernator (EU/US)*/
			|| MATCH_NA("SUPER BOMBERMAN")	/* Super Bomberman 1*/
			|| MATCH_NA("SUPER BOMBERMAN2")	/* Super Bomberman 2*/
			|| MATCH_ID("AS6")		/* Super Bomberman 3*/
			|| MATCH_ID("A4B")		/* Super Bomberman 4*/
			|| MATCH_ID("AYL")		/* Tetris Attack*/
			|| MATCH_NA("POCKY ROCKY")	/* Pocky & Rocky (US/EU)*/
			|| MATCH_NA("KIKIKAIKAI")	/* Kiki Kaikai (JP)*/
			|| MATCH_ID("ANI")		/* Lufia 2 / Estpolis Denki 2*/
			|| MATCH_ID("AQT")		/* Terranigma*/
			|| MATCH_NA("twinbee")		/* Twinbee Rainbow Bell Adventures*/
			|| MATCH_ID("AO7")		/* Tactics Ogre*/
			|| MATCH_NA("Ogre Battle USA")	/* Ogre Battle (US)*/
			|| MATCH_ID("AQ3")		/* Dragon Quest 3*/
			|| MATCH_NA("DRAGONQUEST5")	/* Dragon Quest 5*/
			|| MATCH_ID("AQ6J")		/* Dragon Quest 6*/
			|| MATCH_NA("DARIUS FORCE")	/* Darius Force*/
			|| MATCH_ID("AGC")		/* Front Mission*/
			|| MATCH_ID("AZGJ")		/* Front Mission Gun Hazard*/
			|| MATCH_NA("GANBARE GOEMON")	/* Ganbare Goemon (JP)*/
			|| MATCH_NA("GANBARE GOEMON 2")	/* Ganbare Goemon 2*/
			|| MATCH_NA("mystical ninja")	/* Legend of Mystical Ninja (US/EU)*/
			|| MATCH_NA("NOSFERATU")		/* Nosferatu*/
			|| MATCH_NA("PAC ATTACK")	/* Pac Attack*/
			|| MATCH_NA("PARODIUS")		/* Parodius 1*/
			|| MATCH_NA("PRINCE OF PERSIA")	/* Prince of Persia*/
			|| MATCH_NA("FINAL FANTASY 5")	/* Final Fantasy 5*/
			|| MATCH_NA("OUT OF THIS WORLD") /* Out Of This World (US)*/
			|| MATCH_NA("OUTER WORLD")	/* Outer World (JP)*/
			|| MATCH_NA("ANOTHER WORLD")	/* Another World (EU)*/
			|| MATCH_ID("ABZE")		/* Ballz 3D*/
			|| MATCH_ID("AXSP")		/* Winter Gold*/
			|| MATCH_ID("APJJ")		/* Wonder Project J*/
			|| MATCH_NA("KRUSTYS SUPER FUNHOUSE")	/* Krustys Super Fun House*/
			|| MATCH_NA("KRUSTYS SUPERFUNHOUSE")	/* Krustys Super Fun House*/
			|| MATCH_NA("LEMMINGS")		/* Lemmings 1*/
			|| MATCH_ID("A3Z")		/* Ultimate Mortal Kombat 3*/
			|| MATCH_ID("ARFJ")		/* Star Ocean*/
			|| MATCH_ID("AXBE")		/* Bahamut Lagoon*/
			|| MATCH_ID("ARQ")		/* Realm */
			|| MATCH_NA("S O S")		/* SOS */
			|| MATCH_NA("SUPER DONKEY KONG") /* Super Donkey Kong */
			|| MATCH_ID("AC6J")		/* Cu-On-Pa*/
			|| MATCH_ID("ASR")		/* Street Racer*/
			|| MATCH_NA("SUPER WIDGET")	/* Super Widget*/
			|| MATCH_NA("WILD GUNS")	/* Wild Guns */
			|| MATCH_NA("R-TYPE 3")		/* R-Type 3*/
			|| MATCH_NA("THE NINJAWARRIORS") /* Ninja Warriors */
			|| MATCH_ID("ARW")		/* Super Mario RPG*/
			|| MATCH_NA("SHVC FIREEMBLEM")	/* Fire Emblem - Monshou no Nazo*/
			|| MATCH_ID("BFRJ")		/* Fire Emblem - Thracia 776*/
			|| MATCH_ID("A32J")		/* Fire Emblem - Seisen no Keifu*/
			|| MATCH_ID("AR9")		/* Primal Rage*/
			|| MATCH_ID("APUE")		/* Prehistorik Man*/
			|| MATCH_ID("ALSJ")		/* Lady Stalker*/
			|| MATCH_NA("ROCKMAN&FORTE")	/* Rockman & Forte*/
			|| MATCH_NA("SUPER SWIV")	/* Super SWIV*/
			|| MATCH_NA("CONTRA3 THE ALIEN WARS")	/* Contra 3 The Alien Wars*/
			|| MATCH_NA("EARTHWORM JIM 2")	/* Earthworm Jim 2*/
			|| MATCH_NA("CHOHMAKAIMURA")	/* Chou Makai Mura*/
			|| MATCH_NA("SUPER GHOULS'N GHOSTS")	/* Super Ghouls 'n Ghosts*/
			|| MATCH_NA("X-KALIBER 2097")	/* X-Kaliber 2097*/
			|| MATCH_NA("PHALANX")		/* Phalanx */
			|| MATCH_NA("FINAL FIGHT 2")	/* Final Fight 2*/
			|| MATCH_NA("SUPER TURRICAN 2")	/* Super Turrican 2*/
			|| MATCH_NA("DUNGEON MASTER")	/* Dungeon Master*/
			|| MATCH_NA("DOOM TROOPERS")	/* Doom Troopers*/
			|| MATCH_NA("XAK 1")		/* Xak 1*/
			|| MATCH_NA("XARDION")		/* Xardion*/
			)
		  	PPU.RenderSub = FALSE;
		else
			PPU.RenderSub = TRUE;

		#ifdef __LIBRETRO__
		fprintf(stderr, "PPU.RenderSub = %d\n", PPU.RenderSub);
		#endif`

Problem is, i have no idea what PPU.RenderSub is, is this something they totally added on their end?

**Comment by bearoso** [comment]
That's a nasty, nasty, nasty hack, and won't ever be added to Snes9x mainline.

**Comment by joepogo** [comment]
ok bearoso, thanks for the response! Do you know a list of games you could disable the hi-res with and it wont affect it gameplay-wise? I've been looking for a list. 

And yea, i prefer accuracy over speed, but also speed when accuracy isn't compromised. I was not sure if this was something viable or something not worth the time. :)

**Comment by joepogo** [comment]
Didn't OV fix the Speedy Gonzalez game with this commit? If not, what does this fix?

https://github.com/snes9xgit/snes9x/commit/ef683d61c63088e8a4ccaf47d255f3664b38a421

**Comment by OV2** [comment]
The IRQ improvements fixed issues (that I know of) in jungle book, looney toons, tetris attack, brawl brothers.

**Comment by jcdenton2k** [comment]
I'm going to do a quick test of Air Strike Patrol and see if this works.
I don't have the patience to go through hours of that Speedy Gonzalez game to test it out though.

Will edit this post with results.

**Comment by bearoso** [comment]
Air Strike Patrol won't have the shadow. Heck, only byuu's accuracy branch has the shadow, and that requires a pixel-based PPU emulator, not a scanline-based one, meaning: ultra slow.

Speedy Gonzales won't work, either. That's an HDMA thing, not an IRQ thing.

**Comment by roflcopter777** [comment]
Speedy Gonzales isn't that great of a game IMO, so people can use another emulator for that; not worth sacrificing speed for Air Strike Patrol really.

**Comment by Atavic** [comment]
@joepogo I played _Bomber Man_ for years with no issues.

**Comment by Bachmann1234** [comment]
Howdy. Is this still considered a bug? I have been encountering the super punch out issue running in libretro core lr-SNES9x210

I saw @OV2 's comment about AllowInvalidVRAMAccess being set to TRUE fixing it and how that's the default. But I have not changed any settings. Honestly i dont even know how to set this setting in a libretro context. 

I am skilled enough to get to mad clown consistently and trigger the bug (though as others have said it seems a bit random as you can beat him without triggering it). Is there anything I can do to help track this down?

**Comment by roflcopter777** [comment]
May I ask why you're using Snes9x2010? Snes9x main is much better.

**Comment by Bachmann1234** [comment]
hmm, well that's what was loaded by default. Ill look into upgrading. Thanks

**Comment by roflcopter777** [comment]
There still should be a main Snes9x I'd think?

**Comment by Bachmann1234** [comment]
Yeah, I was able to upgrade without much hassle. Just ran though the clown and no crash. Ill go though it a few more times laster and report back if it still crashes. Thanks !

**Comment by joepogo** [comment]
How so? The bug in super punch out is still there. in snes9x v1.53 it always hangs on the second brain squash with mad clown. It wont do it on the european rom, but it sure will on the us version. Has something been corrected that i am not aware of?

**Comment by Bachmann1234** [comment]
hmm, I did manage to beat him before that...

**Comment by Bachmann1234** [comment]
Just went though it again. Drew the fight out to 2 minutes. Definitely went though his move set a few times. No crash.

**Comment by hatfarm** [comment]
I just installed lr-snes9x tonight (on RetroPie 4.3 on a RPi3) and I was able to repro the crash.  Does not appear to be fixed.  Is there a way to change that setting in an lr release?

**Comment by qwertymodo** [comment]
@hatfarm you'll need to be more specific.  RetroArch maintains several very out-of-date Snes9x forks which exhibit many old bugs.  Exactly which core are you using?

**Comment by hatfarm** [comment]
Hi qwertymodo, this was a response to the post directly above mine. I just installed the latest lr-snes9x from binaries on retropie and the crash is still there for punch out. I would be totally willing to update the config as mentioned further up (the dma one), but I couldn't find where it is in my file system.

**Comment by fr500** [comment]
it only crashes in the NTSC release, it doesn't in the PAL release that could be the reason for the crash.

**Comment by fr500** [comment]
![image](https://user-images.githubusercontent.com/1721040/36360283-66c4129c-14f0-11e8-8f70-ef9f1ac3fec0.png)

This is from the latest release. Savestate:
https://files.andressm.org/f/c1337532b49a4a25ba9e/

Just get a few secs into the fight until he does one of his special moves

**Comment by fr500** [comment]
The savestate is before entering the fight OFC

**Comment by furrykef** [comment]
The Mad Clown bug is almost certainly unrelated to the other issues, so I've created a separate issue just for it (#285).

**Comment by Chris-Halle** [comment]
Snes9x freezes also using game genie and par codes. Found a fix. Revert to old school ZSNES for roms that snes9x will not play or freezes. I switched to it to play Super Punch Out It played like a dream. No crashes, used my codes, all played perfect. Snes9x needs to take a page fro ZSNES on how to make emulators that do not freeze up. MY opinion ZNES is better then Snex9x and it has not been updated since 2007
